packages/core-app-api/src/lib/AuthConnector/PinnipedAuthConnector.ts

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  AuthProviderInfo,
+  ConfigApi,
+  DiscoveryApi,
+} from '@backstage/core-plugin-api';
+import { showLoginPopup } from '../loginPopup';
+import { AuthConnector, CreateSessionOptions, PopupOptions } from './types';
+
+let warned = false;
+
+type Options<AuthSession> = {
+  /**
+   * DiscoveryApi instance used to locate the auth backend endpoint.
+   */
+  discoveryApi: DiscoveryApi;
+  /**
+   * Environment hint passed on to auth backend, for example 'production' or 'development'
+   */
+  environment: string;
+  /**
+   * Information about the auth provider to be shown to the user.
+   * The ID Must match the backend auth plugin configuration, for example 'google'.
+   */
+  provider: AuthProviderInfo;
+  /**
+   * Function used to transform an auth response into the session type.
+   */
+  sessionTransform?(response: any): AuthSession | Promise<AuthSession>;
+  /**
+   * ConfigApi instance used to configure authentication flow of pop-up or redirect.
+   */
+  configApi?: ConfigApi;
+  /**
+   * Options used to configure auth popup
+   */
+  popupOptions?: PopupOptions;
+  audience: string;
```

Comment:
I think it's important to call out this property, since this is the most important distinction between this connector and the default one
```suggestion
  /**
   * Audience parameter associated to sessions
   */
  audience: string;
```